### PR TITLE
Make the search reactive

### PR DESCRIPTION
**Pull request by sventschui**
- [x] Fix styling of `search` block (and `header__meta__search` element)
- [ ] Refactor the model - @davidknezic could you have a look please?
- [ ] Show/Hide suggestions on focus/blur

### CODE REVIEW

**Comment by TheDadi** [comment]
Just because i can...I do not agree with the use of react.js even for components for the Showcase, because once we said we do not want a Framework/Library like react/angular/whatever... in the StyleGuide. :-1:

**Comment by sventschui** [comment]
As you mentioned, it's not in the style guide itself. It's simply for the docs. Your argument is invalid ;)

**Comment by sventschui** [comment]
And to kill your argument at all: React.js is not a framework ;) :laughing:

**Comment by davidknezic** [comment]
I like the approach, I'd even like to build the docs community features as react components.

However, we strictly don't want to replace our example components with react, so we don't require specific framework knowledge in order to be able to use the style guide.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,6 @@
   },
   "browserify-shim": {
     "jquery": "global:jQuery",
-    "baconjs": "global:Bacon",
```
Comment (davidknezic) [inline package.json]: Why not shimmed anymore?

File: docs/js/search/Suggestion.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import classNames from 'classnames';
+
+export default class Suggestion extends React.Component {
+    render() {
```
Comment (davidknezic) [inline docs/js/search/Suggestion.jsx]: 4 space tabs?

**Comment by sventschui** [comment]
@davidknezic When using react we could still provide a data api (as I use in this case too). 

The advantage over jquery would be more structured, simplified components.

The disadvantage is the size (30 vs. 41 KB) and the dependency to polyfills for old browsers.

Maybe it's worth thinking of moving to react entirely. I mean writing a jQuery plugin required "framework skills" (and is VERY UGLY tough)

**Comment by davidknezic** [comment]
Had to switch to node v5 in order for the `npm install` to work, was it the same for you @sventschui?

**Comment by davidknezic** [comment]
Configured `browserify-shim` to do its job globally and shimmed Bacon, React again.

**Comment by davidknezic** [comment]
I sometimes get this weird error:

screen shot 2015-11-18 at 00 33 10

Strangely, everything still works fine, even the Search! Had the same thing when it was `extends React.Component` and wasn't imported like this `import React, { Component } from 'react'`. Any hints @sventschui?

**Comment by sventschui** [comment]
I'm on 5.0.0 anyways ;)

Thanks for having a look at the shim thingy.

Never had this issue but feel free to change the import/extends statements

**Comment by davidknezic** [comment]
Just so it's clear. People will hate us because they'll have to upgrade their `node` because of that. We shouldn't forget to mention that in the changelog.

**Comment by sventschui** [comment]
Have you examined why we need to upgrade? What was the problem in your case?

**Comment by davidknezic** [comment]
Not yet. Weirdly, the `npm install` crashed.

**Comment by davidknezic** [comment]
This [works](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#notable-enhancements) since React 0.14: :tada: 

```jsx
<svg className="search__icon search--header icon-svg">
  <use xlinkHref="${this.props.icon}"/>
</svg>
```

And you know @sventschui that you can do this sort of stuff...

```jsx
__html: '<use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="' + this.props.icon + '"></use>'
```

... using string interpolation...

```jsx
__html: `<use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="${this.props.icon}"></use>`
```

**Comment by davidknezic** [comment]
A lot has changed since last December in React world. What do you think about using redux here @sventschui? I think it's worth closing this pr and making the search more maintainable, since it's central to the style guide.

**Comment by sventschui** [comment]
Agree

**Comment by davidknezic** [comment]
Oh sorry, by closing this pr I meant finishing it. Will have a look at it sometime soon.

**Comment by AndyOGo** [comment]
@sventschui @davidknezic 
Is this ready to be released?

**Comment by sventschui** [comment]
Nope we should use redux instead of Bacon.

@davidknezic is there any possiblity to use the axa.ch search?

**Comment by davidknezic** [comment]
@sventschui Yes we can! :)

**Comment by davidknezic** [comment]
I'll close this and move our search from axa.ch!

**Pull request closed without merging**